src/server/server_hub.go

```diff
@@ -21,16 +22,61 @@ var upgrader = websocket.Upgrader{
 	WriteBufferSize: 1024,
 }
 
+// Main server class.  Should be initialized with NewHub() 
 type ServerHub struct {
```

Comment:
Comment says this is a class, so it should stick to the single responsibility principle. This is way too many fields for a good class, so give some thought to the design here and see whether it can be better abstracted.